packages/network/src/helpers/AddressTools.ts

```diff
@@ -0,0 +1,27 @@
+import ipaddr from 'ipaddr.js'
+
+// IPv4 private address ranges as specified by RFC 1918
+const IPv4PrivateRanges = [
+    '10.0.0.0/8',
+    '172.16.0.0/12',
+    '192.168.0.0/16',
+].map((a: string) => ipaddr.parseCIDR(a))
+
+export function isPrivateIPv4(address: string): boolean {
+    if (ipaddr.IPv4.isValid(address)) {
+        const ip = ipaddr.IPv4.parse(address)
+        for (const range of IPv4PrivateRanges) {
+            if (ip.match(range)) {
+                return true
+            }
+        }
+    }
+
+    return false
+}
+
+export function getAddressFromIceCandidate(candidate: string): string | undefined {
+    const fields = candidate.split(' ').filter((field: string) => field.length > 0)
```

Comment:
note that new versions of typescript can now (probably) infer the `:string` in `.filter((field: string) => field.length > 0`
```suggestion
    const fields = candidate.split(' ').filter((field) => field.length > 0)
```
Doesn't matter much either way, just FYI.

Reply:
Sure :ok_hand: